pkg/version/version.go

```diff
@@ -0,0 +1,41 @@
+package version
+
+import (
+	"github.com/openshift/cluster-samples-operator/pkg/apis/samplesoperator/v1alpha1"
+	"k8s.io/apimachinery/pkg/version"
+)
+
+var (
+	// commitFromGit is a constant representing the source version that
+	// generated this build. It should be set during build via -ldflags.
+	commitFromGit string
+	// versionFromGit is a constant representing the version tag that
+	// generated this build. It should be set during build via -ldflags.
+	versionFromGit string
+	// major version
+	majorFromGit string
+	// minor version
+	minorFromGit string
+)
+
+// Get returns the overall codebase version. It's for detecting
+// what code a binary was built from.
+func Get() version.Info {
```

Comment:
Another option is to just set these into client-go and use that version.  I did that in the router and it simplifies behavior because client go already sends this info to the apiserver and you want that in audit.

See openshift/router makefile

Reply:
oooh the openshift/router way is cool .... switching to that